cmd/remoterun/test.go

```diff
@@ -72,65 +59,49 @@ commands:
 				return err
 			}
 
-			// We need to store the kubeconfig of the current Okteto context locally, so commands
-			// would use the expected
-			kubeconfigPath := getTempKubeConfigFile("test", options.Name)
-			if err := kubeconfig.Write(oktetoContext.GetCurrentCfg(), kubeconfigPath); err != nil {
-				return err
-			}
-			os.Setenv("KUBECONFIG", kubeconfigPath)
-			defer os.Remove(kubeconfigPath)
-
 			dep, err := getDeployable()
 			if err != nil {
-				return fmt.Errorf("could not read information for tests: %w", err)
+				return fmt.Errorf("could not read information to be deployed: %w", err)
 			}
 
 			// Set the default values for the external resources environment variables (endpoints)
 			for name, external := range dep.External {
 				external.SetDefaults(name)
 			}
 
-			runner := &deployable.TestRunner{
-				Executor: executor.NewExecutor(oktetoLog.GetOutputFormat(), false, ""),
-			}
+			k8sClientProvider := okteto.NewK8sClientProviderWithLogger(k8sLogger)
+			cmapHandler := deployCMD.NewConfigmapHandler(k8sClientProvider, k8sLogger)
+
+			runner, err := deployable.NewDeployRunnerForRemote(
+				options.Name,
+				false,
+				cmapHandler,
+				k8sClientProvider,
+				model.GetAvailablePort,
+				k8sLogger,
+			)
 			if err != nil {
 				return fmt.Errorf("could not initialize the command properly: %w", err)
 			}
 
-			os.Setenv(constants.OktetoNameEnvVar, options.Name)
-
-			params := deployable.TestParameters{
-				Name:       options.Name,
-				Namespace:  oktetoContext.GetCurrentNamespace(),
-				Deployable: dep,
-				Variables:  options.Variables,
+			params := deployable.DeployParameters{
+				Name:      options.Name,
+				Namespace: oktetoContext.GetCurrentNamespace(),
+				// For the remote command, the manifest path is the current directory
+				ManifestPath: ".",
+				Deployable:   dep,
+				Variables:    options.Variables,
 			}
 
-			c := &TestCommand{
+			c := &DeployCommand{
```

Comment:
Also, this `DeployCommand` will redeploy the external resources. If we don't share the `OKTETO_ENV` between deploy and test, we might be losing endpoints or failing the operation: https://github.com/okteto/okteto/blob/10d1b4633a369e64462057f190e25f80eb7923ce/pkg/externalresource/external_resource.go#L104